src/authorization/permissions/MethodModesExtractor.ts

```diff
@@ -1,36 +1,46 @@
 import type { Operation } from '../../http/Operation';
+import type { ResourceSet } from '../../storage/ResourceSet';
 import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
 import { ModesExtractor } from './ModesExtractor';
 import { AccessMode } from './Permissions';
 
 const READ_METHODS = new Set([ 'GET', 'HEAD' ]);
-const WRITE_METHODS = new Set([ 'PUT', 'DELETE' ]);
-const APPEND_METHODS = new Set([ 'POST' ]);
-const SUPPORTED_METHODS = new Set([ ...READ_METHODS, ...WRITE_METHODS, ...APPEND_METHODS ]);
+const SUPPORTED_METHODS = new Set([ ...READ_METHODS, 'PUT', 'POST', 'DELETE' ]);
 
 /**
  * Generates permissions for the base set of methods that always require the same permissions.
  * Specifically: GET, HEAD, POST, PUT and DELETE.
  */
 export class MethodModesExtractor extends ModesExtractor {
+  private readonly resourceSet: ResourceSet;
+
+  public constructor(resourceSet: ResourceSet) {
+    super();
+    this.resourceSet = resourceSet;
+  }
+
   public async canHandle({ method }: Operation): Promise<void> {
     if (!SUPPORTED_METHODS.has(method)) {
       throw new NotImplementedHttpError(`Cannot determine permissions of ${method}`);
     }
   }
 
-  public async handle({ method }: Operation): Promise<Set<AccessMode>> {
+  public async handle({ method, target }: Operation): Promise<Set<AccessMode>> {
     const result = new Set<AccessMode>();
     if (READ_METHODS.has(method)) {
       result.add(AccessMode.read);
     }
-    if (WRITE_METHODS.has(method)) {
+    if (method === 'PUT') {
       result.add(AccessMode.write);
+      if (!await this.resourceSet.hasResource(target)) {
+        result.add(AccessMode.create);
+      }
+    }
+    if (method === 'POST') {
```

Comment:
(mainly added the other comments to make this case clear; because `POST` is the only case where the identified thing is the parent container of the actual resource)